Use $CC to link (and check for broken compilers), such that we don't get
linked modules of a different bitness.
http://bugs.gentoo.org/show_bug.cgi?id=297751

--- hints/darwin.sh
+++ hints/darwin.sh
@@ -128,13 +128,13 @@
 ccflags="${ccflags} ${cppflags}"
 
 # Known optimizer problems.
-case "`cc -v 2>&1`" in
+case "`${cc:-gcc} -v 2>&1`" in
   *"3.1 20020105"*) toke_cflags='optimize=""' ;;
 esac
 
 # Shared library extension is .dylib.
 # Bundle extension is .bundle.
-ld='cc';
+ld="${cc:-gcc}";
 so='dylib';
 dlext='bundle';
 usedl='define';
